Gems/PhysX/Code/Tests/Benchmarks/PhysXBenchmarksUtilities.cpp

```diff
@@ -14,20 +14,36 @@
 #include <AzFramework/Physics/Shape.h>
 #include <AzFramework/Physics/PhysicsScene.h>
 #include <AzFramework/Physics/Configuration/RigidBodyConfiguration.h>
+#include <Benchmarks/PhysXBenchmarksCommon.h>
 
 namespace PhysX::Benchmarks
 {
     namespace Utils
     {
-        AzPhysics::SimulatedBodyHandleList CreateRigidBodies(int numRigidBodies,
-            AzPhysics::Scene* scene, bool enableCCD,
+        BenchmarkRigidBodies CreateRigidBodies(
+            int numRigidBodies,
+            AzPhysics::SceneHandle sceneHandle,
+            bool enableCCD,
+            int benchmarkObjectType,
             GenerateColliderFuncPtr* genColliderFuncPtr /*= nullptr*/, GenerateSpawnPositionFuncPtr* genSpawnPosFuncPtr /*= nullptr*/,
             GenerateSpawnOrientationFuncPtr* genSpawnOriFuncPtr /*= nullptr*/, GenerateMassFuncPtr* genMassFuncPtr /*= nullptr*/,
             GenerateEntityIdFuncPtr* genEntityIdFuncPtr /*= nullptr*/
         )
         {
-            AzPhysics::SimulatedBodyHandleList rigidBodies;
-            rigidBodies.reserve(numRigidBodies);
+            BenchmarkRigidBodies benchmarkRigidBodies;
+
+            if (benchmarkObjectType == PhysX::Benchmarks::RigidBodyApiObject)
+            {
+                AzPhysics::SimulatedBodyHandleList rigidBodies;
+                rigidBodies.reserve(numRigidBodies);
+                benchmarkRigidBodies = AZStd::move(rigidBodies);
+            }
+            else if (benchmarkObjectType == PhysX::Benchmarks::RigidBodyEntity)
+            {
+                PhysX::EntityList rigidBodies;
+                rigidBodies.reserve(numRigidBodies);
+                benchmarkRigidBodies = AZStd::move(rigidBodies);
```

Comment:
Why is the rigidBodies vector being reserved and then an empty contained moved into benchmarkRigidBodies?

Reply:
This is to initialize the std variant with the right type of the vector and have that vector with the right pre-allocated capacity.